remove conventional commit

SirSpidey opened the pull request:
### Description

Conventional commits are no longer a requirement.

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

ocofaigh commented on docs/contribute-module.md:
```diff
@@ -111,7 +111,7 @@ For more information about testing modules in this project, see [Tests](tests.md
 After your PR tests pass, create a pull request.
 
-!> **Important**: Use [Conventional Commit](https://www.conventionalcommits.org) messages when you commit code ([cheat sheet](https://cheatography.com/albelop/cheat-sheets/conventional-commits/)). Conventional commits determine whether a new version of a module is needed and which [semantic versioning number](versioning.md) to use. The commit messages are published in the release notes for the mo ...
+When the PR is merged, your commit messages might change because the merging team uses the [squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits) option to merge PRs. The PR commit message determines whether a new version of a module is needed and w ...
```
Perhaps need to more clear here:

`The PR commit message determines whether a new version...` -> `The PR commit message used during the squash and merge determines whether a new version...`

SirSpidey replied:
how about slightly different: 

The PR commit message `that's included by the merging team` determines whether a new vesion...

> When the PR is merged, your commit messages might change because the merging team uses the squash and merge option to merge PRs. The PR commit message that's included by the merging team determines whether a new version of a module is needed and which semantic versioning number to use. The message also is published in the release notes for the module.

ocofaigh replied:
fine with me

ocofaigh commented on docs/implementation-guidelines.md:
```diff
@@ -55,11 +55,10 @@ Modules must contain at least one automated validation test that runs Terraform
 Module changes adhere to [semantic versioning](https://semver.org/), with releases labeled as `{major}.{minor}.{patch}`. [Git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) are used to identify the module versions. For more information about versioning, see [Release versioning](versioning.md).
 
-As a module author or contributor, you don't need to release a new version. Conventional commits determine whether a new version of a module is needed and which semantic versioning number to use. When a PR is merged into the main branch, a new release is created by the release automation if the validation pipeline passes. The commit messages are published in the release notes for the module.
+As a module author or contributor, you don't need to release a new version. The merging team determines whether a new version of a module is needed and which semantic versioning number to use based on the commits in the PR. When a PR is merged into the main branch, a new release is created by the release automation if the validation pipeline passes. The commit messages are published in the releas ...
```
I'm not sure we need any of this...

> The merging team determines whether a new version of a module is needed and which semantic versioning number to use based on the commits in the PR
^ This is not true - the merging team will determine if a version is needed based on the content of the PR, not the commits used. They may also check the checkboxes in the PR description, but that is not always accurately checked by the contributor. 

> When a PR is merged into the main branch, a new release is created by the release automation if the validation pipeline passes.
^ This is not true. There is no validation pipeline when a PR is merge. Just a release pipeline

> The commit messages are published in the release notes for the module.
^ Its actually the commit message used by the merging team when they squash and merge

So I think all of this content can be removed. And perhaps rename the title just `Versioning`

I have an action to create a merger team specific doc with more info on squash and merge process, but its not something that general contributors need to know about.

SirSpidey replied:
OK. Removed that paragraph and the next about conventional commits that slipped by

ocofaigh commented on docs/tests.md:
```diff
@@ -27,7 +27,6 @@ Make sure that you include at least these two tests:
-    - If you must implement a breaking change to the module in a PR, include a Conventional Commit footer in one of the commits with the text `BREAKING CHANGE:` at the beginning of the footer. For more information, see the [Conventional Commits cheat sheet](https://cheatography.com/albelop/cheat-sheets/conventional-commits/).
     - If the change breaks example code but not module code, include `SKIP UPGRADE TEST` in the commit message. Make sure to describe why you are breaking the example in the PR description.
 
   ?> **Tip**: The upgrade test is disabled by default in `pr_test.go` in the module template because you can't run an upgrade test until the initial module code is merged to the main branch. After the initial merge, create a pull request to enable the upgrade test by commenting out the line that starts with `t.Skip`.
```
This is no longer a requirement as the line that starts with `t.Skip` has been removed from the module template (too many people were not removing it, and tests were being skipped). Perhaps we should mention that `SKIP UPGRADE TEST` may need to be used in one of the commit message for a modules first PR since there won't be any pre existing module code to do an upgrade test with.

SirSpidey replied:
how about this?

<img alt="image" src="https://github.com/terraform-ibm-modules/documentation/assets/8194208/8b5b2b2b-e41c-4e76-acdf-d2c5ed196f2c">

SirSpidey commented on docs/tests.md:
```diff
@@ -27,7 +27,6 @@ Make sure that you include at least these two tests:
 
     Use the `RunTestUpgrade()` test helper to help determine compatibility with earlier module versions.
 
-    - If you must implement a breaking change to the module in a PR, include a Conventional Commit footer in one of the commits with the text `BREAKING CHANGE:` at the beginning of the footer. For more information, see the [Conventional Commits cheat sheet](https://cheatography.com/albelop/cheat-sheets/conventional-commits/).
```
@ocofaigh @daniel-butler-irl,
 
> maybe we mention using SKIP UPGRADE TEST instead

like this? 

> Upgrade test
> 
> Use the `RunTestUpgrade()` test helper to help determine compatibility with earlier module versions.
> 
> If you need to skip the test, include `SKIP UPGRADE TEST` in the commit message and `BREAKING CHANGE: ` at the beginning of the footer of the commit message. Include information about the breaking changes.


OK?

ocofaigh replied:
You don't need both. One or the other. So I suggest just `SKIP UPGRADE TEST`. However I think we need to make it clear that if skipping the upgrade test, they need to fully explain in the PR why they are doing it, and what the impact is to consumers. I don't want people skipping upgrade tests because they failed and not investigating why.

SirSpidey replied:
see updated at https://github.com/terraform-ibm-modules/documentation/pull/47#discussion_r1317435050

SirSpidey commented:
@ocofaigh @daniel-butler-irl I think I have addressed the issues. You can see how the published docs will look at: 
- https://pages.github.ibm.com/allen-dean/tim-docs/#/contribute-module?id=commit-your-code-and-submit-for-review
- https://pages.github.ibm.com/allen-dean/tim-docs/#/tests?id=the-pr_testgo-file
- https://pages.github.ibm.com/allen-dean/tim-docs/#/implementation-guidelines?id=versioning
